etl/steps/data/garden/ember/2022-08-01/european_electricity_review.py

```diff
@@ -0,0 +1,276 @@
+    )
+
+    # Create a table with a well-constructed index.
+    table = (
+        catalog.Table(df)
+        .set_index(["country", "year"], verify_integrity=True)
+        .sort_index()
+    )
+
+    return table
+
+
+def process_country_overview_data(table: catalog.Table) -> catalog.Table:
+    """Process country overview data, including country harmonization.
+
+    Parameters
+    ----------
+    table : catalog.Table
+        Table from the meadow dataset.
+
+    Returns
+    -------
+    table: catalog.Table
+        Processed table.
+
+    """
+    # Rename columns for consistency with global electricity review.
+    columns = {
+        "country_name": "country",
+        "year": "year",
+        "generation_twh": "generation__twh",
+        "net_import_twh": "net_imports__twh",
+        "demand_twh": "demand__twh",
+        "demand_mwh_per_capita": "demand_per_capita__kwh",
+    }
+    df = pd.DataFrame(table).reset_index()[list(columns)].rename(columns=columns)
+    # Harmonize country names.
+    df = geo.harmonize_countries(
+        df=df,
+        countries_file=str(COUNTRY_MAPPING_PATH),
+        warn_on_unused_countries=False,
+        warn_on_missing_countries=True,
+    )
+
+    # Convert units of demand per capita.
+    df["demand_per_capita__kwh"] = df["demand_per_capita__kwh"] * MWH_TO_KWH
+
+    # Create a table with a well-constructed index.
+    table = (
+        catalog.Table(df)
+        .set_index(["country", "year"], verify_integrity=True)
+        .sort_index()
+    )
+
+    return table
+
+
+def process_emissions_data(table: catalog.Table) -> catalog.Table:
+    """Process emissions data, including country harmonization.
+
+    Parameters
+    ----------
+    table : catalog.Table
+        Table from the meadow dataset on emissions.
+
+    Returns
+    -------
+    table: catalog.Table
+        Processed table.
+
+    """
+    # Rename columns for consistency with global electricity review.
+    columns = {
+        "country_name": "country",
+        "year": "year",
+        "emissions_intensity_gco2_kwh": "co2_intensity__gco2_kwh",
+        "emissions_mtc02e": "total_emissions__mtco2",
+    }
+    df = pd.DataFrame(table).reset_index()[list(columns)].rename(columns=columns)
+    # Harmonize country names.
+    df = geo.harmonize_countries(
+        df=df,
+        countries_file=str(COUNTRY_MAPPING_PATH),
+        warn_on_unused_countries=False,
+        warn_on_missing_countries=True,
+    )
+
+    # Create a table with a well-constructed index.
+    table = (
+        catalog.Table(df)
+        .set_index(["country", "year"], verify_integrity=True)
+        .sort_index()
+    )
+
+    return table
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load data.
+    #
+    # Read dataset from meadow.
+    ds_meadow = catalog.Dataset(MEADOW_DATASET_PATH)
+
+    # Load countries-regions table (required to convert country codes to country names in net flows table).
+    countries_regions = catalog.find(
```

Comment:
It think it's better to get data with `countries_regions = Dataset(DATA_DIR / 'garden/owid/reference')['countries_regions']` just like meadow when you can. It's not an issue here with countries regions, but it is fetching data from remote catalog and users might expect it to fetch data from local catalog.

This was a long version of saying you shouldn't do this for meadow versions (or upstream deps of this step) and that it's ok for countries regions, although getting it directly is better.

(You can obviously keep it like this)

Reply:
Done